### PR TITLE
Fix localdns, add --no-tunnel

### PR DESCRIPTION
**Pull request by ostenbom**
Local-dns was broke because asking dnsmasq to reload did not get it to reload the config file from start.
Here, we create the config file correctly from the beginning instead. Much nicer.

I have also added a `--no-tunnel` argument, this both helps when cloudflare is down, but also to check if local-dns works 😅

### CODE REVIEW

File: linkup-cli/src/background_booting.rs [hunk 1]
```diff
@@ -31,12 +29,16 @@ pub fn boot_background_services() -> Result<(), CliError> {
...
     wait_till_ok(format!("{}linkup-check", local_url))?;
 
-    if is_tunnel_started().is_err() {
+    if state.should_use_tunnel() && is_tunnel_started().is_err() {
         println!("Starting tunnel...");
         let tunnel = start_tunnel()?;
-        state.linkup.tunnel = tunnel;
-    } else {
+        state.linkup.tunnel = Some(tunnel);
+    } else if state.should_use_tunnel() {
```
Comment (jonaslindstr) [inline linkup-cli/src/background_booting.rs:36]: This condition is repeated twice; I think you could check for it once and move the other condition inside that If-statement as it's a little hard to follow the logic.

File: linkup-cli/src/main.rs [hunk 2]
```diff
@@ -149,7 +156,13 @@ enum LocalDNSSubcommand {
 #[derive(Subcommand)]
 enum Commands {
     #[clap(about = "Start a new linkup session")]
-    Start,
+    Start {
+        #[clap(
+            long = "no-tunnel",
```
Comment (jonaslindstr) [inline linkup-cli/src/main.rs:161]: How about a short option alias?

File: linkup-cli/src/services/dnsmasq.rs [hunk 3]
```diff
@@ -14,16 +10,16 @@ const CONF_FILE: &str = "dnsmasq-conf";
...
 
     Command::new("dnsmasq")
         .current_dir(linkup_dir_path())
         .arg("--log-queries")
         .arg("-C")
         .arg(conf_file_path)
-        .stdout(Stdio::null())
-        .stderr(Stdio::null())
+        // .stdout(Stdio::null())
+        // .stderr(Stdio::null())
```
Comment (jonaslindstr) [inline linkup-cli/src/services/dnsmasq.rs:22]: Commented out code

Comment (ostenbom) [reply linkup-cli/src/services/dnsmasq.rs:22]: Ah thanks missed that 😅

File: linkup-cli/src/main.rs (hunk 2 above)
Comment (jonaslindstr) [reply linkup-cli/src/main.rs:161]: Almost all options are free, so `-n` would do the job.

**Review by jonaslindstr — APPROVED** [review]
